include/circt/Dialect/ESI/ESIManifest.td

```diff
@@ -112,6 +113,10 @@ def ServiceImplRecordOp : ESI_Op<"esi.manifest.service_impl", [
     qualified($appID) (`svc` $service^)? `by` $serviceImplName `with` $implDetails
     attr-dict-with-keyword $reqDetails
   }];
+
+  let extraClassDeclaration = [{
+    void getDetails(SmallVectorImpl<NamedAttribute> &results);
```

Comment:
'getDetails' is pretty vague to me - referencing the summary, does this mean that the details are the "information necessary to connect to the service and service clients"? - it probably should have a comment.

Reply:
This method is part of the IsMetadata OpInterface and is described there...  in not too much more detail.